commons/ava_services/gecko_service_initializer_core.go

```diff
@@ -98,14 +146,85 @@ func (g GeckoServiceInitializerCore) GetStartCommand(publicIpAddr string, depend
 		for _, service := range avaDependencies {
 			socket := service.GetStakingSocket()
 			socketStrs = append(socketStrs, fmt.Sprintf("%s:%d", socket.GetIpAddr(), socket.GetPort().Int()))
+			if g.stakingTlsEnabled {
+				// We hardcode the first bootstrapper ID from the TLS identities in gecko_service_tls_identities
+				commandList = append(commandList, "--bootstrap-ids=" + STAKER_1_NODE_ID)
+				// You only need one bootstrapper dependency, and we only currently have one cert -> ID mapping so break for loop here.
+				break
+			}
 		}
 		joinedSockets := strings.Join(socketStrs, ",")
 		commandList = append(commandList, "--bootstrap-ips=" + joinedSockets)
 	}
 	logrus.Debugf("Command list: %+v", commandList)
-	return commandList
+	return commandList, nil
 }
 
 func (g GeckoServiceInitializerCore) GetServiceFromIp(ipAddr string) services.Service {
 	return GeckoService{ipAddr: ipAddr}
 }
+
+
+func getServiceCertAndKeyFiles(cert *x509.Certificate, ca *x509.Certificate) (certFile *bytes.Buffer, keyFile *bytes.Buffer, err error) {
+	certPrivKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "")
+	}
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, ca, &certPrivKey.PublicKey, certPrivKey)
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "")
+	}
+	certPEM := new(bytes.Buffer)
+	pem.Encode(certPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: certBytes,
+	})
+
+	certPrivKeyPEM := new(bytes.Buffer)
+	pem.Encode(certPrivKeyPEM, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(certPrivKey),
+	})
+	return certPEM, certPrivKeyPEM, nil
+}
+
+func getRootCA() *x509.Certificate {
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2020),
+		Subject: pkix.Name{
+			Organization:  []string{"Kurtosis Technologies"},
+			Country:       []string{"USA"},
+			Province:      []string{""},
+			Locality:      []string{""},
+			StreetAddress: []string{""},
+			PostalCode:    []string{""},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+	return ca
+}
+
+func getServiceCert() *x509.Certificate {
+	cert := &x509.Certificate{
+		SerialNumber: big.NewInt(1993),
```

Comment:
From https://ldapwiki.com/wiki/Certificate%20Serial%20Number#:~:text=The%20Certificate%20Serial%20Number%20field,contain%20the%20same%20serial%20number: "An Certificate Issuer must ensure that no two distinct Certificates with the same Certificate Issuer DN contain the same serial number."

Since we can have N many dependent nodes, can we generate the SerialNumber randomly so we don't get collisions? Technically the cert's hash would still be unique due to the `NotBefore` and `NotAfter` fields, but reallllllly those shouldn't be dynamically generated in this function (and instead should be passed in, for consistency, so all certs have exactly the same)

Reply:
👍

Reply:
done